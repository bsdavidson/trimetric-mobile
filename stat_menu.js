import React, {Component} from "react";
import {
  StyleSheet,
  Text,
  Image,
  View,
  TouchableOpacity,
  Switch,
  ScrollView
} from "react-native";
import {connect} from "react-redux";
import LayersMenu from "./layers_menu";
import helpIcon from "./assets/layers.png";
import busIcon from "./assets/bus.png";
import stopIcon from "./assets/stop.png";
import {updateLayerVisibility} from "./actions";
import {getVehiclePoints, getSelectedItem} from "./selectors";

class StatMenu extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false
    };

    this.handleTogglePress = this.handleTogglePress.bind(this);
  }

  handleTogglePress() {
    this.setState({visible: !this.state.visible});
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.selectedItem) {
      this.setState({visible: false});
    }
  }

  render() {
    return (
      <View
        style={[
          styles.container,
          {
            height: this.state.visible ? "40%" : 30
          }
        ]}>
        <View style={styles.headerContainer}>
          <Text>
            <Image style={styles.busImage} source={busIcon} />:{" "}
            {this.props.activeVehicles &&
              this.props.activeVehicles.features.length}
          </Text>
          <TouchableOpacity
            style={styles.closeButton}
            onPress={this.handleTogglePress}>
            {this.state.visible ? (
              <Text style={styles.closeButtonIcon}>&#x00D7;</Text>
            ) : (
              <Text style={styles.menuButtonIcon}>&#9776;</Text>
            )}
          </TouchableOpacity>
        </View>
        <ScrollView style={{marginTop: 20}}>
          <View style={styles.stat}>
            <Text style={styles.statTitle}>Help</Text>
            <View style={{flex: 0, flexDirection: "row", maxWidth: 280}}>
              <Image style={styles.busImage} source={busIcon} />
              <Text style={styles.statText}>Bus Stats</Text>
            </View>
            <View style={{flex: 0, flexDirection: "row"}}>
              <Image style={styles.stopImage} source={stopIcon} />
              <Text style={styles.statText}>Stop Stats</Text>
            </View>
          </View>
        </ScrollView>
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    layerVisibility: state.layerVisibility,
    activeVehicles: getVehiclePoints(state),
    selectedItem: getSelectedItem(state)
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onUpdateLayerVisibility: (layerName, value) => {
      dispatch(updateLayerVisibility(layerName, value));
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(StatMenu);

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: "#FFFFFF",
    shadowColor: "#000",
    shadowOffset: {width: 0, height: -2},
    shadowOpacity: 0.2,
    shadowRadius: 2,
    zIndex: 51
  },
  headerContainer: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    height: 30,
    backgroundColor: "#FFFFFF",
    borderBottomWidth: 1,
    borderBottomColor: "#cccccc",
    zIndex: 50
  },

  stat: {
    flex: 0,
    marginTop: 0,
    marginLeft: 0,
    padding: 10
  },
  statTitle: {
    color: "#AAAAAA",
    fontSize: 19,
    marginBottom: 10,
    marginLeft: 25
  },
  statText: {
    fontSize: 14,
    marginBottom: 15
  },
  busImage: {
    height: 20,
    width: 20,
    margin: 5,
    marginTop: 2
  },
  stopImage: {
    height: 30,
    width: 20,
    margin: 5,
    marginTop: 2
  },
  closeButton: {
    flex: 0,
    position: "absolute",
    right: 0,
    zIndex: 99
  },
  closeButtonIcon: {
    color: "#AAAAAA",
    fontSize: 28,
    fontWeight: "bold",
    lineHeight: 28,
    margin: 0,
    marginRight: 8,
    padding: 0,
    paddingLeft: 6,
    paddingRight: 6
  },
  menuButtonIcon: {
    fontSize: 18,
    marginRight: 16,
    marginTop: 2,
    color: "#888888"
  }
});
