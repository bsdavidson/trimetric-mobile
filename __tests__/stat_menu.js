import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import {StatMenu} from "../stat_menu";

it("renders stat bar", () => {
  const tree = renderer.create(<StatMenu activeVehicles={1} />).toJSON();
  expect(tree).toMatchSnapshot();
});
