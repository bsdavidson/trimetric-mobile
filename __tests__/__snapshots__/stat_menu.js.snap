// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders stat bar 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#34495e",
        "bottom": 0,
        "height": 40,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": -2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
        "zIndex": 51,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#ecf0f1",
        "fontSize": 16,
        "padding": 8,
        "paddingLeft": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      1
    </Text>
     
    active vehicles
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 0,
        "opacity": 1,
        "position": "absolute",
        "right": 5,
        "zIndex": 99,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#EFEFEF",
          "fontFamily": "Allerta Stencil",
          "fontSize": 22,
          "marginRight": 20,
          "marginTop": 4,
        }
      }
    >
      ?
    </Text>
  </View>
</View>
`;
